lib/stages/rfiBroadcast.cpp

```diff
@@ -228,6 +233,12 @@ void rfiBroadcast::main_thread() {
             mask_percent_metric.labels({std::to_string(current_freq_bin)})
                 .set(perc_zeroed.average());
 
+            // Increment the prometheus metrics for the total number of samples and the total number
+            // of flagged samples
+            sample_counter.labels({std::to_string(current_freq_bin)})
+                .inc((double)_samples_per_data_set / _sk_step);
```

Comment:
The double cast seems out of place here.

Reply:
Do you not need it for rounding errors?

Reply:
I think `_samples_per_data_set` is always and exactly multiple of `_sk_step` otherwise none of the RFI kernels would work. This just casts it do a double, does a division and then will implicitly cast it back to an int.

Also this quantity never changes so I would define it as a `const uint64_t` at the top of the main thread.

Reply:
At the very least I would pull it to the top and make the cast back to an int deliberate. In the case they don't exactly divide, it's not case what this value should actually be, so at least if you do that you can make it clear.